### ISSUE TITLE
Truck Wheels clipping to surface. Vehicles stuck on side.

### ISSUE THREAD

**Issue by GrimJerr** [shadowmage45/KerbalFoundries2#55]
![screenshot15](https://user-images.githubusercontent.com/29767895/78471136-8d1ff100-76f4-11ea-827e-778583ec6b1d.png)

**Comment by GrimJerr** [comment]
Love the Mod but I am having a lot of frustration with the wheels clipping to the ground when the vehicle flips or rolls

**Comment by GrimJerr** [comment]
https://youtu.be/D9DSomCp5cs
You can see it in this video also @ 12:06
Sometimes you get lucky and get unstuck, but mostly it is just stuck, respawn.

**Comment by shadowmage45** [comment]
Yeah, you've managed to find the one issue of the wheel sim that I've never found a solution for.  Basically, I use colliders as proxies to keep the wheels from over-compressing.  Those same colliders have to be set to be frictionless or it causes problems at max compression.

But then... when the vehicle goes on its side, you end up with the issue seen above. (as it ends up resting on these colliders, and slides all over because they are frictionless)

I wonder... can I make a spherical collider, and then scale it down on the x-axis?  so it becomes a weird 3d elipsoid?

**Comment by GrimJerr** [comment]
I don't have the software or the experience but I was envisioning a dome on the end of the Hub with colliders, that would allow it to roll back onto one of the other axis. If it was modeled without a texture it would be invisible (?), of course then you'd run the risk of running into other objects with an invisible Collider on the end of the hub.

**Comment by shadowmage45** [comment]
> of course then you'd run the risk of running into other objects with an invisible Collider on the end of the hub.

Yep, that would be the issue.  But also likely to be the solution, or something similar.  

As long as the additional colliders were positioned and sized appropriately, they could be kept within the bounds of the visible wheel meshes, and so should line up closely with what would be expected.

Notably, I need to see if I can scale a sphere collider on one axis.... (as noted above).

Should be solvable, but might take a bit of thought and careful manipulation...
